src/System.Linq.Parallel/tests/WithCancellationTests.cs

```diff
@@ -14,24 +14,23 @@ namespace System.Linq.Parallel.Tests
     // the tests here are only regarding basic API correctness and sanity checking.
     public static class WithCancellationTests
     {
-        [Fact]
-        public static void MultiplesWithCancellationIsIllegal()
+        [Theory]
+        [MemberData(nameof(Sources.Ranges), new[] { 16 }, MemberType = typeof(Sources))]
```

Comment:
Why do we need to do this for all of these different sources?  Seems like unnecessary duplication.  This is a test of WithCancellation, not of how WithCancellation interacts with various sources and operators other than WithCancellation.